src/DynamoCoreWpf/Extensions/IViewExtension.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Dynamo.Extensions;
+
+namespace Dynamo.Wpf.Extensions
+{
+    /// <summary>
+    /// An extension to the UI layer of Dynamo.
+    /// </summary>
+    public interface IViewExtension : IDisposable
```

Comment:
They look similar, but it's only skin deep.  The `XParams` object handed to the various methods i completely different and expected to contain completely different types of fields.

Now, I could do something like:

```
public interface IExtension<StartType, ReadyType> : IDisposable
```

Then have something like:

```
public interface IExtensionManager<ExtensionType> : IDisposable where ExtensionType : IExtension<StartType, ReadyType>
```

But, at first sight the loss in readability isn't really worth the effort.  

These are tiny interfaces and the `Params` types are the only ones I expect to really change much.